crates/task-impls/src/consensus.rs

```diff
@@ -456,10 +457,10 @@ where
 
                         let leaf: SequencingLeaf<_> = SequencingLeaf {
                             view_number: view,
-                            height: proposal.height,
+                            height: proposal.block_header.block_number,
                             justify_qc: proposal.justify_qc.clone(),
                             parent_commitment,
-                            deltas: Right(proposal.block_commitment),
+                            deltas: Right(proposal.block_header.commitment),
```

Comment:
Do we want this to be a commitment?  Where else can we retrieve the block header information from if we need it later?

Reply:
+1 the header should be small enough to keep in the leaf/decide event

Reply:
This is bad naming. 😅 What's stored here is the _payload_ commitment, which can be retrieved later if we want to recover the block _header_. I'll update it to `block_header.payload_commitment()`.

Reply:
Oh, I probably misunderstood both of your suggestions. Did you mean we should not store the commitment, but use the block header directly? If so, I'm not sure whether to keep the `height` field. If `delta` is `Left`, it won't contain the height info, but we'll have redundant height if it's `Right`.

Reply:
On second thought, this may not cause redundancy. I could change the `Either` type to be either a tuple of the block number and payload, or a block header. I'll try this and hopefully this won't cause much type complexity.